### PR TITLE
Fixed #35022 -- Fixed RenameIndex() crash on unnamed indexes if exists unique constraint on the same fields.

### PR DESCRIPTION
**Pull request by David-Wobrock**
[Ticket 35022](https://code.djangoproject.com/ticket/35022)

### CODE REVIEW

File: tests/migrations/test_operations.py [hunk 1]
```diff
@@ -3626,6 +3626,22 @@ def test_rename_index_unknown_unnamed_index(self):
...
+            unique_together=True,
+        )
+        operation = migrations.RenameIndex(
+            "Pony", new_name="new_pony_test_idx", old_fields=["pink", "weight"]
+        )
+        new_state = project_state.clone()
+        operation.state_forwards(app_label, new_state)
+        # Rename index.
+        with connection.schema_editor() as editor:
+            operation.database_forwards(app_label, editor, project_state, new_state)
```
Comment (felixxm) [inline tests/migrations/test_operations.py:3643]: Let's add an assertion to confirm that is actually works:
```suggestion
            operation.database_forwards(app_label, editor, project_state, new_state)
        self.assertIndexNameExists(table_name, "new_pony_test_idx")
```

File: tests/migrations/test_operations.py [hunk 2]
```diff
@@ -3626,6 +3626,22 @@ def test_rename_index_unknown_unnamed_index(self):
             with self.assertRaisesMessage(ValueError, msg):
                 operation.database_forwards(app_label, editor, project_state, new_state)
 
+    def test_rename_index_unnamed_index_with_unique_index(self):
+        app_label = "test_rninunuwui"
```
Comment (felixxm) [inline tests/migrations/test_operations.py:3630]: :wink: 
```suggestion
        app_label = "test_rninuniwui"
```

Comment (David-Wobrock) [reply tests/migrations/test_operations.py:3630]: _Of course, how could I miss that_ 🤣

File: tests/migrations/test_operations.py [hunk 3]
```diff
@@ -3626,6 +3626,26 @@ def test_rename_index_unknown_unnamed_index(self):
             with self.assertRaisesMessage(ValueError, msg):
                 operation.database_forwards(app_label, editor, project_state, new_state)
 
+    @skipUnlessDBFeature("allows_multiple_constraints_on_same_fields")
```
Comment (felixxm) [inline tests/migrations/test_operations.py:3629]: I added `skipUnlessDBFeature` as I realized it won't work on Oracle.

**Review by felixxm — COMMENTED** [review]
@David-Wobrock Thanks :+1:

File: tests/migrations/test_operations.py (hunk 3 above)
Comment (David-Wobrock) [reply tests/migrations/test_operations.py:3629]: Good catch, thanks 🙏

**Pull request merged**